### PR TITLE
Fix regression in getOperationAndEllipsoidFromProjString when used in QgsCoordinateReferenceSystem::syncDatabase()

### PR DESCRIPTION
**Pull request by agiudiceandrea**
## Description

Since https://github.com/qgis/QGIS/commit/9e0bf6df993a8548be3e21e274ee2857728e658d (https://github.com/qgis/QGIS/pull/40737), `QgsCoordinateReferenceSystem::syncDatabase()` doesn't properly insert [about 100 srs definitions](https://gist.github.com/agiudiceandrea/b4a9f191f0a4b4b7f5126868e6eac722) into tbl_srs because `getOperationAndEllipsoidFromProjString` fails to properly retrieve ellipsoid name from proj string and doesn't satisfy the NOT NULL constraint of ellipsoid_acronym field in tbl_srs.

This fixes [about 100 errors](https://gist.github.com/agiudiceandrea/b4a9f191f0a4b4b7f5126868e6eac722) like:

> After a 'make install' I received sevaral similar messages, please see a
> cut below:
> 
> 
> Could not execute: INSERT INTO tbl_srs(srs_id,
> description,projection_acronym,ellipsoid_acronym,parameters,srid,auth_name,auth_id,is_geo,deprecated)
> VALUES (63422, 'NAD27 (CRS27)','longlat',NULL,'+proj=longlat
> +datum=NAD27 +no_defs',520003422,'OGC','CRS27',1,0) [NOT NULL constraint
> failed: tbl_srs.ellipsoid_acronym/NOT NULL constraint failed:
> tbl_srs.ellipsoid_acronym]
> 
> Could not execute: INSERT INTO tbl_srs(srs_id,
> description,projection_acronym,ellipsoid_acronym,parameters,srid,auth_name,auth_id,is_geo,deprecated)
> VALUES (63423, 'NAD83 (CRS83)','longlat',NULL,'+proj=longlat
> +datum=NAD83 +no_defs',520003423,'OGC','CRS83',1,0) [NOT NULL constraint
> failed: tbl_srs.ellipsoid_acronym/NOT NULL constraint failed:
> tbl_srs.ellipsoid_acronym]
>
> ...
> ...
> ...

See https://lists.osgeo.org/pipermail/qgis-user/2021-June/049055.html

### CODE REVIEW

**Comment by agiudiceandrea** [comment]
@m-kuhn could you please have a look at this?

File: src/core/proj/qgscoordinatereferencesystem.cpp [hunk 1]
```diff
@@ -2134,7 +2133,7 @@ void getOperationAndEllipsoidFromProjString( const QString &proj, QString &opera
     // and will result in oddities within other areas of QGIS (e.g. project ellipsoid won't be correctly
     // set for these CRSes). Better just hack around and make the constraint happy for now,
     // and hope that the definitions get corrected in future.
-    ellipsoid.clear();
+    ellipsoid = "";
```
Comment (m-kuhn) [inline src/core/proj/qgscoordinatereferencesystem.cpp:2136]: What's this for?

Comment (agiudiceandrea) [reply src/core/proj/qgscoordinatereferencesystem.cpp:2136]: to ensure the returned ellipsoid is not NULL but and empty string eventually.

Comment (agiudiceandrea) [reply src/core/proj/qgscoordinatereferencesystem.cpp:2136]: this reverst https://github.com/qgis/QGIS/commit/9e0bf6df993a8548be3e21e274ee2857728e658d#diff-0eb6084517fa5891245c56de95f3c07dc2424ffc25b732059acce2bdcfb6c66fL2461-R2480

Comment (m-kuhn) [reply src/core/proj/qgscoordinatereferencesystem.cpp:2136]: Does this void the not null constraint? In this case, is the not null constraint justified?

Comment (agiudiceandrea) [reply src/core/proj/qgscoordinatereferencesystem.cpp:2136]: I don't know. See the comment:
```
    // satisfy not null constraint on ellipsoid_acronym field
    // possibly we should drop the constraint, yet the crses with missing ellipsoid_acronym are malformed
    // and will result in oddities within other areas of QGIS (e.g. project ellipsoid won't be correctly
    // set for these CRSes). Better just hack around and make the constraint happy for now,
    // and hope that the definitions get corrected in future.
```

Comment (agiudiceandrea) [reply src/core/proj/qgscoordinatereferencesystem.cpp:2136]: Anyway, probably there aren't CRSs **without** ellipsoid/datum name in proj strings retrieved from PROJ database.

The relevant fix is to substitute
```c++
  if ( !ellipseMatch.hasMatch() )
  {
    ellipsoid = ellipseMatch.captured( 1 );
  }
```
with
```c++
  if ( ellipseMatch.hasMatch() )
  {
    ellipsoid = ellipseMatch.captured( 1 );
  }
```
to ensure that CRSs **with** ellipsoid/datum name in proj string will be properly imported.

**Review by nyalldawson — APPROVED** [review]
LGTM

**Pull request merged**